test/tests/functional/pbs_offline_vnodes.py

```diff
@@ -374,10 +355,151 @@ def test_fail_action_startup_hook(self):
+        Since it is the only mom, all vnodes reported by her
+        should also be offline.
+        """
+        single_mom = ""
+        single_mom = self.moms.values()[0]
+        self.create_multi_vnodes(1)
+        self.server.expect(NODE, {ATTR_NODE_state: 'free'},
+                           id=single_mom.shortname, max_attempts=3,
+                           interval=2)
+
+        mom_host = single_mom.shortname
+        pbs_exec = self.server.pbs_conf['PBS_EXEC']
+        pbsnodes_cmd = os.path.join(pbs_exec, 'bin', 'pbsnodes')
+        pbsnodes_offline = pbsnodes_cmd + ' -o ' + mom_host
+        self.du.run_cmd(self.server.hostname, cmd=pbsnodes_offline, sudo=True)
+
+        # the mom node and all of her children should be offline
+        self.server.expect(
+            NODE, {ATTR_NODE_state: 'offline'},
+            id=single_mom.shortname, max_attempts=20, interval=2)
+
+        vname = None
+        if single_mom.is_cpuset_mom() is True:
+            vnl = self.server.status(NODE)
+            for v1 in vnl:
+                if single_mom.shortname != v1['resources_available.vnode']:
+                    if vname is None:
+                        vname = v1['resources_available.vnode']
+                    # Check that the vnode is in offline state
+                    self.server.expect(
+                        VNODE, {'state': 'offline'},
+                        id=v1['resources_available.vnode'], max_attempts=3,
+                        interval=2)
+        else:
+            vname = self.verify_other_vnodes_state('offline')
+
+    def test_pbsnodes_o_multi_mom_only_one_offline(self):
+        """
+        Offline one mom using pbsnodes -o.
+        In the case of multiple moms reporting the same set of vnodes,
+        none of the vnodes should be marked offline,
+        including the children vnodes.
+        NOTE: This test needs moms to report the same set of vnodes.
+        """
+        if len(self.moms.values()) != 2:
+            self.skipTest("Provide 2 moms while invoking test")
+
+        if self.moms.values()[0].is_cpuset_mom() is True or self.moms.values()[
+                1].is_cpuset_mom() is True:
+            self.skipTest("Skipping test on cpuset moms")
+
+        momA = ""
+        momA = self.moms.values()[0]
+        momB = ""
+        momB = self.moms.values()[1]
+
+        # The moms provided to the test may have unwanted vnodedef files.
+        if momA.has_vnode_defs():
+            momA.delete_vnode_defs()
+        if momB.has_vnode_defs():
+            momB.delete_vnode_defs()
+
+        self.create_multi_vnodes(2)
+
+        # Offline only one of the moms, the other mom and her children
+        # should still be free
+        pbs_exec = self.server.pbs_conf['PBS_EXEC']
+        pbsnodes_cmd = os.path.join(pbs_exec, 'bin', 'pbsnodes')
+        pbsnodes_offline = pbsnodes_cmd + ' -o ' + momA.shortname
+        self.du.run_cmd(self.server.hostname, cmd=pbsnodes_offline, sudo=True)

```

Comment:
run_cmd() works better with a list of arguments, not a string. Pass pbsnodes_offline = [pbsnodes_cmd, '-o', momA.shortname]